src/nnet3/nnet-descriptor-test.cc

```diff
@@ -217,6 +217,46 @@ void UnitTestGeneralDescriptorSpecial() {
   KALDI_ASSERT(NormalizeTextDescriptor(names, "Offset(a, 0)") == "a");
 }
 
+int32 GetInputIntervalFromString(const std::vector<std::string> &node_names,
+                                 const std::string &desc_str,
+                                 const std::string &input_name) {
+  std::vector<std::string> tokens;
+  DescriptorTokenize(desc_str, &tokens);
+  tokens.push_back("end of input");
+  const std::string *next_token = &(tokens[0]);
+  GeneralDescriptor *gen_desc = GeneralDescriptor::Parse(node_names,
+                                                         &next_token);
+  if (*next_token != "end of input") 
+    KALDI_ERR << "Parsing Descriptor, expected end of input but got "
+              << "'" <<  *next_token << "'";
+  return GetInputIntervalInternal(*gen_desc, node_names, input_name);
+}
+
+void UnitTestGetInputIntervalSpecial() {
+  std::vector<std::string> names;
+  names.push_back("a");
+  names.push_back("b");
+  names.push_back("c");
+  names.push_back("d");
+  KALDI_ASSERT(GetInputIntervalFromString(names, "ReplaceIndex(a, t, 0)", "a")
+               == 0);
+  KALDI_ASSERT(GetInputIntervalFromString(names, "ReplaceIndex(a, t, 10)", "a")
+               == 1);
+  KALDI_ASSERT(GetInputIntervalFromString(names, "Round(a, 10)", "a") == 10);
+  KALDI_ASSERT(GetInputIntervalFromString(names, "Round(Round(a, 10), 5)", "a")
```

Comment:
We will never have a chance to have t=5 since the inner Round makes it impossible